AutomationHelpers/src/UserCodeCollections/FileLibrary.cs

```diff
@@ -152,9 +153,169 @@ public static void WaitForFile(string path, string pattern, int duration, int in
             }
         }
 
+        /// <summary>
+        /// Compares content of two binary files.
+        /// </summary>
+        /// <param name="filePath1">The relative or absolute path of the first file</param>
+        /// <param name="filePath2">The relative or absolute path of the second file</param>
+        [UserCodeMethod]
+        public static void CompareBinaryFiles(string filePath1, string filePath2)
+        {
+            try
+            {
+                filePath1 = GetPathForFile(filePath1);
+                filePath2 = GetPathForFile(filePath2);
+
+                if (!FilesExist(filePath1, filePath2))
+                {
+                    return;
+                }
+
+                var fi1 = new FileInfo(filePath1);
+                var fi2 = new FileInfo(filePath2);
+
+                if (fi1.Length != fi2.Length)
+                {
+                    Report.Failure("Files '" + filePath1 + "' and '" + filePath2 + "' are not equal.");
+                    return;
+                }
+
+                int bufferSize = 65536;
+
+                using (var fs1 = fi1.OpenRead())
+                using (var fs2 = fi2.OpenRead())
+                using (var bs1 = new BufferedStream(fs1, bufferSize))
+                using (var bs2 = new BufferedStream(fs2, bufferSize))
+                {
+                    var tempBufferSize = sizeof(Int64);
+                    var tempBuffer1 = new byte[tempBufferSize];
+                    var tempBuffer2 = new byte[tempBufferSize];
+
+                    for (var i = 0L; i < (fi1.Length / tempBufferSize) + 1; i++)

```

Comment:
I think your algorithm is ok, although for the last iteration the tempBuffers will not be full and comparison will also be done on bytes not actually read in that iteration. So it will compare bytes from current and previous iteration - anyway, only if all iterations have same content, the validation succeeds, which is what we want.

Reply:
Bytes from last iteration will not have impact on current iteration. They have to be equal to reach that iteration at all. So, only bytes from current iteration will make difference if any. We can clean buffers after comparison, but it seems to me as unnecessary operation.